dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1694,15 +1696,19 @@ private boolean sameJavaSpecVersion() {
     private boolean areConfiguredFeaturesGood(Set<String> newConfiguredFeatures, Set<String> newConfiguredPlatforms) {
         if (!!!featureRepository.isDirty()
             && !!!featureRepository.hasConfigurationError()
-            && featureRepository.getConfiguredFeatures().equals(newConfiguredFeatures)
-            && featureRepository.getPlatforms().equals(newConfiguredPlatforms)) {
-            // check that all installed features are still installed
-            for (String resolvedFeature : featureRepository.getResolvedFeatures()) {
-                if (featureRepository.getFeature(resolvedFeature) == null) {
-                    return false;
+            && featureRepository.getConfiguredFeatures().equals(newConfiguredFeatures)){
+            if( !isBeta || 
+                (isBeta && featureRepository.getPlatforms().equals(newConfiguredPlatforms) 
+                && ((featureRepository.getPlatformEnvVar() == null && platformEnvironmentVariable == null) ||
+                (featureRepository.getPlatformEnvVar() != null && featureRepository.getPlatformEnvVar().equals(platformEnvironmentVariable))))) {
+                    
+                // check that all installed features are still installed
```

Comment:
This block seems expensive?     Is this checking every feature installed in this local server?